src/App.jsx

```diff
@@ -14,33 +17,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [activeButton, setActiveButton] = useState(null);
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortAlph = () => {
+    const sorted = [...visibleGoods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (isReversed) {
+      sorted.sort().reverse();
+    } else {
+      sorted.sort();
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    setVisibleGoods(sorted);
+    setActiveButton('Sort alphabetically');
+  };
+
+  const sortLength = () => {
+    const sorted = [...visibleGoods];
+
+    if (isReversed) {
+      sorted.sort((a, b) => b.length - a.length);
+    } else {
+      sorted.sort((a, b) => a.length - b.length);
```

Comment:
The `sortLength` function correctly sorts by length, but ensure that the logic for reversing is consistent with the `sortAlph` function. If you decide to adjust the `sortAlph` logic, consider applying similar changes here for consistency.